cfme/utils/appliance/__init__.py

```diff
@@ -1327,6 +1327,56 @@ def update_rhel(self, *urls, **kwargs):
 
         return result
 
+    def upgrade(self, upgrade_to, cfme_only=True, reboot=False):
+        """ Upgrade an appliance
+        Args:
+            upgrade_to (str): Stream for appliance upgrade. supported (5.9.z, 5.10.z, 5.11.z)
+            cfme_only (bool): Update cfme packages only or all available updates.
+            reboot (bool): Reboot appliance after upgrade.
+
+        Note: Appliance always upgrades to latest available version.
+        In supported stream `.z`indicate latest available.
+        """
+
+        logger.info("Appliance upgrade process started")
+
+        if self.version.series() not in upgrade_to:
+            # TODO: Add support of major upgrade
+            raise NotImplementedError("Major upgrade not supported")
+
+        supported_stream_repo_map = {
+            "5.9.z": "update_url_59",
+            "5.10.z": "update_url_510",
+            "5.11.z": "update_url_511",
+        }
+        try:
+            update_url = conf.cfme_data["basic_info"][supported_stream_repo_map[upgrade_to]]
+        except (KeyError, IndexError):
+            raise ValueError("Need to specify the correct upgrade path in cfme_data.yaml")
+
+        logger.debug("Adding update repo to appliance")
+        result = self.ssh_client.run_command(f"curl {update_url} -o /etc/yum.repos.d/update.repo")
+        logger.debug(result.output)
+
+        self.evmserverd.stop()
+
+        logger.info("Running yum update")
+        cmd = " cfme" if cfme_only else ""
+        try:
+            result = self.ssh_client.run_command(f"yum -y update{cmd}", timeout=3600)
+        except socket.timeout:
+            logger.error(f"SSH timed out while updating appliance: {result.output}")
+
+        self.db_service.restart()
+        self.evmserverd.start()
+
+        # May be chance to update kernel with all update.
+        if reboot or not cfme_only:
+            self.reboot()
```

Comment:
@digitronik could we move the reboot before the db and evmserver service restarts/start? Like:
```
 # May be chance to update kernel with all update.
 if reboot or not cfme_only:
     self.reboot()
else:
    self.db_service.restart()
    self.evmserverd.start()
```
Those services will be restarted with a reboot anyway. Thoughts?

Reply:
sure :+1: